aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CreateHandler.java

```diff
@@ -16,35 +18,61 @@
             final ResourceHandlerRequest<ResourceModel> request,
             final CallbackContext callbackContext,
             final ProxyClient<RdsClient> proxyClient,
             final RequestLogger requestLogger) {
-        final ResourceModel model = request.getDesiredResourceState();
-        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
-                .then(progress -> {
-                    if (StringUtils.isNullOrEmpty(model.getDBParameterGroupName()))
-                        model.setDBParameterGroupName(IdentifierUtils.generateResourceIdentifier(
-                                request.getStackId(),
-                                Optional.ofNullable(request.getLogicalResourceIdentifier()).orElse("dbparametergroup"),
-                                request.getClientRequestToken(),
-                                MAX_LENGTH_GROUP_NAME
-                        ).toLowerCase());
-                    return ProgressEvent.progress(model, progress.getCallbackContext());
-                })
-                .then(progress -> proxy.initiate("rds::create-db-parameter-group", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
-                        .translateToServiceRequest(resourceModel -> Translator.createDbParameterGroupRequest(resourceModel, Tagging.mergeTags(request.getSystemTags(), request.getDesiredResourceTags())))
-                        .backoffDelay(CONSTANT)
-                        .makeServiceCall((createDBParameterGroupRequest, proxyInvocation) ->
-                                proxyInvocation.injectCredentialsAndInvokeV2(createDBParameterGroupRequest, proxyInvocation.client()::createDBParameterGroup))
-                        .handleError((createDBParameterGroupRequest, exception, client, resourceModel, ctx) ->
-                                Commons.handleException(
-                                        ProgressEvent.progress(resourceModel, ctx),
-                                        exception,
-                                        DEFAULT_DB_PARAMETER_GROUP_ERROR_RULE_SET))
-                        .done((paramGroupRequest, paramGroupResponse, proxyInvocation, resourceModel, context) -> applyParameters(proxy, proxyInvocation, resourceModel, context, requestLogger)))
+
+        final ResourceModel desiredModel = request.getDesiredResourceState();
+
+        final Tagging.TagSet extraTags = Tagging.TagSet.builder()
+                .stackTags(Tagging.translateTagsToSdk(request.getDesiredResourceTags()))
+                .resourceTags(new HashSet<>(Translator.translateTagsToSdk(request.getDesiredResourceState().getTags())))
+                .build();
+
+        return ProgressEvent.progress(desiredModel, callbackContext)
+                .then(progress -> setDBParameterGroupNameIfEmpty(request, desiredModel, progress))
+                .then(progress -> createDBParameterGroup(proxy, proxyClient, progress, request.getSystemTags(), requestLogger))
```

Comment:
I think I would prefer to pack all tags in a single-type container, even though this container will only deliver a set of system tags. This approach will make `createDBParameterGroup` more generic to what exactly came its way.